src/views/docs/en/guides/extend/plugins.md

```diff
@@ -154,20 +157,108 @@ rule-two
+
+Architect provides a suite of runtime helpers via the [`@architect/functions`][functions] library. This library leverages functionality provided by [`deploy`][deploy] and [`sandbox`][sandbox] to expose runtime variables enabling [service discovery][discovery] - the automatic configuration, search and discovery of infrastructure and services making up your application. The `variables` plugin method enables plugin authors to hook into the Architect service discovery mechanism.
+
+This method is used by Architect in two situations:
+
+1. When running in a local development context via [`sandbox`][sandbox], `sandbox` will invoke the `variables` method in order to compile all runtime variables required by Architect application plugins and provide them to [`@architect/functions`][functions].
+2. When running [`arc deploy`][deploy], it will invoke the `variables` method in order to compile an [AWS SSM Parameter][ssm] per variable exported by the method. Later, when running in a remotely-deployed runtime context on AWS Lambda, [`@architect/functions`][functions] will query the [AWS SSM Parameter Store][ssm] to retrieve these variables at runtime.
+
+The `variables` plugin method is only necessary to implement if you would like your plugin to provide runtime data within Lambdas via the [`@architect/functions`][functions] library. The exported variables would be available via the [`services` function][services] provided by [`@architect/functions`][functions] (namespaced under the plugin name). For more information on how to query the service discovery mechanism using [`@architect/functions`][functions] at runtime, check out the [`@architect/functions` `services` documentation][services].
+
+#### Arguments
+
+All arguments arrive as a bag of options with the following properties:
+
+|Argument|Description|
+|---|---|
+|`arc`|Object representing the [parsed Architect project manifest](https://github.com/architect/parser) file for the current project|
+|`cloudformation`|The [CloudFormation JSON template][cfn-ref] making up the Architect project|
+|`stage`|The name of the environment; usually one of `testing`, `staging` or `production`; `testing` is provided when running in a [`sandbox`][sandbox] context whereas `staging` and `production` are provided at Architect CLI runtime when either `arc deploy staging` or `arc deploy production` are invoked, respectively|
+|`inventory`|An [Architect inventory object][inv] representing the current Architect project|
+
+#### Returns
+
+This method should always return an object. Each property on the object represents a variable name, and the value for each property contains the variable value.
+
+> 🏌️‍♀️ Protip: When this method is invoked in a pre-`deploy` context, acceptable values for the variables include CloudFormation JSON. This is essential to expose CloudFormation-managed infrastructure.
```

Comment:
This is a little fuzzy to me, maybe we can flesh this out a bit more, provide an exaple/use-case, etc.?

Reply:
OK, I will add 'see the example below' where one can see this point in action.